src/client/Microsoft.Identity.Client.Brokers/RuntimeBroker.cs

```diff
@@ -0,0 +1,313 @@
+        }
+
+        public async Task<MsalTokenResponse> AcquireTokenSilentDefaultUserAsync(
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenSilentParameters acquireTokenSilentParameters)
+        {
+            var cancellationToken = authenticationRequestParameters.RequestContext.UserCancellationToken;
+            MsalTokenResponse msalTokenResponse = null;
+
+            _logger.Verbose("[WamBroker] Acquiring token silently for default account.");
+
+            using (var core = new NativeInterop.Core())
+            using (var authParams = WamAdapters.GetCommonAuthParameters(authenticationRequestParameters, _wamOptions.MsaPassthrough))
+            {
+                using (NativeInterop.AuthResult result = await core.SignInSilentlyAsync(
+                        authParams,
+                        authenticationRequestParameters.CorrelationId.ToString("D"),
+                        cancellationToken).ConfigureAwait(false))
+                {
+                    if (result.IsSuccess)
+                    {
+                        msalTokenResponse = WamAdapters.ParseRuntimeResponse(result, authenticationRequestParameters, _logger);
+                    }
+                    else
+                    {
+                        WamAdapters.ThrowExceptionFromWamError(result, authenticationRequestParameters, _logger);
+                    }
+                }
+            }
+
+            return msalTokenResponse;
+        }
+
+        public async Task RemoveAccountAsync(ApplicationConfiguration appConfig, IAccount account)
+        {
+            string correlationId = Guid.NewGuid().ToString();
+
+            //if OperatingSystemAccount is passed then we use the user signed-in on the machine
+            if (PublicClientApplication.IsOperatingSystemAccount(account))
+            {
+                throw new MsalException("wam_remove_account_failed", "Could not remove the default os account");
+            }
+
+            _logger.Info($"Removing WAM Account. Correlation ID : {correlationId} ");
+
+            using (var core = new NativeInterop.Core())
+            {
+                using (var wamAccount = await core.ReadAccountByIdAsync(
+                    account.HomeAccountId.ObjectId,
+                    correlationId).ConfigureAwait(false))
+                {
+                    if (wamAccount == null)
+                    {
+                        _logger.WarningPii(
+                            $"Could not find a WAM account for the selected user {account.Username}",
+                            "Could not find a WAM account for the selected user");
+                    }
+                    
+                    using (NativeInterop.SignOutResult result = await core.SignOutSilentlyAsync(
+                        appConfig.ClientId,
+                        correlationId,
+                        wamAccount).ConfigureAwait(false))
```

Comment:
What if `wamAccount` is null here?

Reply:
throwing exception if wamAccount is null